selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/preferences/ContributeTabTest.java

```diff
@@ -0,0 +1,288 @@
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.NotificationsPopupPanel;
+import org.eclipse.che.selenium.pageobject.PanelSelector;
+import org.eclipse.che.selenium.pageobject.Preferences;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.PullRequestPanel;
+import org.eclipse.che.selenium.pageobject.git.Git;
+import org.eclipse.che.selenium.refactor.move.MoveItemsTest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Aleksandr Shmaraev */
+public class ContributeTabTest {
+  private static final Logger LOG = LoggerFactory.getLogger(MoveItemsTest.class);
+  private static final String FIRST_PROJECT_NAME = "first-vcs-project";
+  private static final String SECOND_PROJECT_NAME = "second-vcs-project";
+  private static final String THIRD_PROJECT_NAME = "not-vcs-project";
+  private static final String EXP_TEXT_NOT_VCS =
+      "Project does not provide VCS, supported by contribution plugin.";
+  private static final String NOTIFICATION_MESS =
+      "To activate Contribute Panel by clicking on a project go to Profile -> Preferences -> Git -> Contribute";
+
+  private String firstProjectUrl;
+  private String secondProjectUrl;
+
+  @Inject private Ide ide;
+  @Inject private Menu menu;
+  @Inject private Git git;
+  @Inject private DefaultTestUser user;
+  @Inject private AskDialog askDialog;
+  @Inject private Preferences preferences;
+  @Inject private TestWorkspace testWorkspace;
+  @Inject private TestGitHubRepository testRepo;
+  @Inject private TestGitHubRepository testRepo2;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private PullRequestPanel pullRequestPanel;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private NotificationsPopupPanel notificationsPopupPanel;
+  @Inject private PanelSelector panelSelector;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
+    testRepo.addContent(entryPath);
+    testRepo2.addContent(entryPath);
+
+    URL resource = getClass().getResource("/projects/guess-project");
+    testProjectServiceClient.importProject(
+        testWorkspace.getId(),
+        Paths.get(resource.toURI()),
+        THIRD_PROJECT_NAME,
+        ProjectTemplates.MAVEN_SPRING);
+
+    ide.open(testWorkspace);
+  }
+
+  @AfterMethod
+  public void closeForm() {
+    try {
+      if (askDialog.isOpened()) {
+        askDialog.confirmAndWaitClosed();
+      }
+
+      if (preferences.isPreferencesFormOpened()) {
+        preferences.clickOnCloseButton();
+      }
+
+      preferences.waitPreferencesFormIsClosed();
+
+    } catch (Exception e) {
+      LOG.error(e.getLocalizedMessage(), e);
```

Comment:
If it's an error, we should fail the test. Otherwise it's better to log is a warning.